packages/runtime/src/useCases/consumption/attributes/DeleteRepositoryAttribute.ts

```diff
@@ -0,0 +1,70 @@
+import { AccountController, CoreId } from "@nmshd/transport";
+import { Inject } from "typescript-ioc";
+import { AttributeIdString, RuntimeErrors, SchemaRepository, SchemaValidator, UseCase } from "../../common";
+
+export interface DeleteRepositoryAttributeRequest {
+    attributeId: AttributeIdString;
+}
+
+class Validator extends SchemaValidator<DeleteRepositoryAttributeRequest> {
+    public constructor(@Inject schemaRepository: SchemaRepository) {
+        super(schemaRepository.getSchema("DeleteRepositoryAttributeRequest"));
+    }
+}
+
+export class DeleteRepositoryAttributeUseCase extends UseCase<DeleteRepositoryAttributeRequest, void> {
+    public constructor(
+        @Inject private readonly attributeController: AttributesController,
+        @Inject private readonly accountController: AccountController,
+        @Inject validator: Validator
+    ) {
+        super(validator);
+    }
+
+    protected async executeInternal(request: DeleteRepositoryAttributeRequest): Promise<Result<void>> {
+        const repositoryAttributeId = CoreId.from(request.attributeId);
+        const repositoryAttribute = await this.attributeController.getLocalAttribute(repositoryAttributeId);
+
+        if (typeof repositoryAttribute === "undefined") {
+            return Result.fail(RuntimeErrors.general.recordNotFound(LocalAttribute));
+        }
+
+        if (!repositoryAttribute.isRepositoryAttribute(this.accountController.identity.address)) {
+            return Result.fail(RuntimeErrors.attributes.isNotRepositoryAttribute(repositoryAttributeId));
+        }
+
+        const ownSharedIdentityAttributes = await this.attributeController.getLocalAttributes({ "shareInfo.sourceAttribute": repositoryAttributeId.toString() });
+        const ownSharedIdentityAttributePredecessors = await this.attributeController.getSharedPredecessorsOfRepositoryAttribute(repositoryAttribute);
+        for (const ownSharedAttribute of [...ownSharedIdentityAttributes, ...ownSharedIdentityAttributePredecessors]) {
+            if (!ownSharedAttribute.isOwnSharedAttribute(this.accountController.identity.address)) {
+                throw new Error(
+                    `The Attribute ${ownSharedAttribute.id} does not fulfill the requirements of an own shared Attribute, so it cannot be adjusted adequately, deleting its source Attribute.`
+                );
+            }
+
+            ownSharedAttribute.shareInfo.sourceAttribute = undefined;
+            await this.attributeController.updateAttributeUnsafe(ownSharedAttribute);
+        }
+
+        if (typeof repositoryAttribute.succeededBy !== "undefined") {
+            const successor = await this.attributeController.getLocalAttribute(repositoryAttribute.succeededBy);
+            if (typeof successor === "undefined") {
+                throw new Error(`The Attribute ${repositoryAttribute.succeededBy} was not found, even though it is specified as successor of Attribute ${repositoryAttribute.id}.`);
+            }
```

Comment:
this should also be checked BEFORE changing any attributes to make sure to not corrupt the data